capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -175,6 +189,131 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     String requestResultJSON = gson.toJson(getRequestResult(request));
     response.getWriter().println(requestResultJSON);
   }
-  //TODO doPost for creating and storing a study set
 
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
+    HashMap<String, Object> result_par = new Gson()
+    .fromJson(request.getReader(), HashMap.class);
+    String ownderID = result_par.get("user_id").toString();
```

Comment:
You can't trust the `user_id` field here. I can use curl to send a request to your API with a different user's ID. Instead you need to use the User API in Java to get the email of the currently logged in user and update your SQL queries to select the user ID for that email. You should be able to do a nested SELECT within the INSERT statement.